launch/launch/actions/include_launch_description.py

```diff
@@ -160,8 +161,13 @@ def execute(self, context: LaunchContext) -> List[LaunchDescriptionEntity]:
             perform_substitutions(context, normalize_to_list_of_substitutions(arg_name))
             for arg_name, arg_value in self.launch_arguments
         ]
-        declared_launch_arguments = (
-            launch_description.get_launch_arguments_with_include_launch_description_actions())
+        try:
+            declared_launch_arguments = (
+                launch_description.get_launch_arguments_with_include_launch_description_actions())
+        except Exception as exc:
+            if hasattr(exc, 'add_note'):
+                exc.add_note(f'while executing {self.describe()}')
```

Comment:
Unfortunately, mypy is still complaining about `add_note()`. It seems like it doesn't understand the `hasattr()` check:

```
201 files checked
2 errors
launch/actions/include_launch_description.py: note: In member "execute" of class "IncludeLaunchDescription":
launch/actions/include_launch_description.py:169:17: error: "Exception" has no attribute "add_note"
Found 1 error in 1 file (checked 201 source files)
```

https://ci.ros2.org/job/ci_linux-rhel/2027/consoleText

This was apparently fixed in mypy version 0.991: https://github.com/python/mypy/issues/1424. RHEL unfortunately uses 0.971 (search for "mypy==0.971" in the full CI job output).

At this point, since this won't fail at runtime, I would just ignore mypy for this line, which I _assume_ should do the trick:

```suggestion
                exc.add_note(f'while executing {self.describe()}')  # type: ignore
```

Same for the other file, although I wonder why mypy isn't complaining about it.

Sorry this is taking so much time.